From e28a1d4efd591fd0f663104f51ca37b0a844206a Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 14 Nov 2016 18:22:09 +0800
Subject: [PATCH 1274/5242] MLK-13405-2 ARM: dts: add usdhc2 support for
 imx6sll-lpddr3-arm2

commit  d28521a7ffbf127aadc1abf540fb0818bf3a68c6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add usdhc2 support for imx6sll-lpdr3-arm2 board.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts |   74 ++++++++++++++++++++++-------
 1 file changed, 58 insertions(+), 16 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts b/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
index dc5d745..c524816 100644
--- a/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
@@ -109,11 +109,10 @@
 
 		reg_sd2_vmmc: sd2_vmmc {
 			compatible = "regulator-fixed";
-			regulator-name = "SD2_SPWR";
-			regulator-min-microvolt = <3000000>;
-			regulator-max-microvolt = <3000000>;
-			gpio = <&gpio4 27 GPIO_ACTIVE_HIGH>;
-			enable-active-high;
+			regulator-name = "eMMC_VCCQ";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-boot-on;
 		};
 
 		reg_sd3_vmmc: sd3_vmmc {
@@ -407,7 +406,7 @@
 				MX6SLL_PAD_KEY_ROW7__GPIO4_IO07 0x17059
 				MX6SLL_PAD_GPIO4_IO22__GPIO4_IO22 0x17059
 				MX6SLL_PAD_KEY_COL3__GPIO3_IO30	0x17059
-				MX6SLL_PAD_SD2_RESET__GPIO4_IO27 0x17059
+				MX6SLL_PAD_SD2_RESET__GPIO4_IO27 0x417059
 				MX6SLL_PAD_KEY_COL4__GPIO4_IO00 0x17059
 				MX6SLL_PAD_REF_CLK_32K__GPIO3_IO22 0x17059 /* SD3 CD */
 				MX6SLL_PAD_KEY_COL6__GPIO4_IO04 0x17059 /*SD3 RESET */
@@ -579,19 +578,51 @@
 
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
-				MX6SLL_PAD_SD2_CMD__SD2_CMD	0x17059
-				MX6SLL_PAD_SD2_CLK__SD2_CLK	0x17059
-				MX6SLL_PAD_SD2_DATA0__SD2_DATA0 0x17059
-				MX6SLL_PAD_SD2_DATA1__SD2_DATA1	0x17059
-				MX6SLL_PAD_SD2_DATA2__SD2_DATA2	0x17059
-				MX6SLL_PAD_SD2_DATA3__SD2_DATA3	0x17059
-				MX6SLL_PAD_SD2_DATA4__SD2_DATA4	0x17059
-				MX6SLL_PAD_SD2_DATA5__SD2_DATA5	0x17059
-				MX6SLL_PAD_SD2_DATA6__SD2_DATA6	0x17059
-				MX6SLL_PAD_SD2_DATA7__SD2_DATA7	0x17059
+				MX6SLL_PAD_SD2_CMD__SD2_CMD		0x17059
+				MX6SLL_PAD_SD2_CLK__SD2_CLK		0x13059
+				MX6SLL_PAD_SD2_DATA0__SD2_DATA0 	0x17059
+				MX6SLL_PAD_SD2_DATA1__SD2_DATA1		0x17059
+				MX6SLL_PAD_SD2_DATA2__SD2_DATA2		0x17059
+				MX6SLL_PAD_SD2_DATA3__SD2_DATA3		0x17059
+				MX6SLL_PAD_SD2_DATA4__SD2_DATA4		0x17059
+				MX6SLL_PAD_SD2_DATA5__SD2_DATA5		0x17059
+				MX6SLL_PAD_SD2_DATA6__SD2_DATA6		0x17059
+				MX6SLL_PAD_SD2_DATA7__SD2_DATA7		0x17059
+				MX6SLL_PAD_GPIO4_IO21__SD2_STROBE	0x413059
+			>;
+		};
+
+		pinctrl_usdhc2_100mhz: usdhc2grp_100mhz {
+			fsl,pins = <
+				MX6SLL_PAD_SD2_CMD__SD2_CMD		0x170b9
+				MX6SLL_PAD_SD2_CLK__SD2_CLK		0x130b9
+				MX6SLL_PAD_SD2_DATA0__SD2_DATA0 	0x170b9
+				MX6SLL_PAD_SD2_DATA1__SD2_DATA1		0x170b9
+				MX6SLL_PAD_SD2_DATA2__SD2_DATA2		0x170b9
+				MX6SLL_PAD_SD2_DATA3__SD2_DATA3		0x170b9
+				MX6SLL_PAD_SD2_DATA4__SD2_DATA4		0x170b9
+				MX6SLL_PAD_SD2_DATA5__SD2_DATA5		0x170b9
+				MX6SLL_PAD_SD2_DATA6__SD2_DATA6		0x170b9
+				MX6SLL_PAD_SD2_DATA7__SD2_DATA7		0x170b9
+				MX6SLL_PAD_GPIO4_IO21__SD2_STROBE	0x4130b9
 			>;
 		};
 
+		pinctrl_usdhc2_200mhz: usdhc2grp_200mhz {
+			fsl,pins = <
+				MX6SLL_PAD_SD2_CMD__SD2_CMD		0x170f9
+				MX6SLL_PAD_SD2_CLK__SD2_CLK		0x130f9
+				MX6SLL_PAD_SD2_DATA0__SD2_DATA0 	0x170f9
+				MX6SLL_PAD_SD2_DATA1__SD2_DATA1		0x170f9
+				MX6SLL_PAD_SD2_DATA2__SD2_DATA2		0x170f9
+				MX6SLL_PAD_SD2_DATA3__SD2_DATA3		0x170f9
+				MX6SLL_PAD_SD2_DATA4__SD2_DATA4		0x170f9
+				MX6SLL_PAD_SD2_DATA5__SD2_DATA5		0x170f9
+				MX6SLL_PAD_SD2_DATA6__SD2_DATA6		0x170f9
+				MX6SLL_PAD_SD2_DATA7__SD2_DATA7		0x170f9
+				MX6SLL_PAD_GPIO4_IO21__SD2_STROBE	0x4130f9
+			>;
+		};
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX6SLL_PAD_SD3_CMD__SD3_CMD	0x17059
@@ -724,6 +755,17 @@
 	status = "okay";
 };
 
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+	vqmmc-supply = <&reg_sd2_vmmc>;
+	bus-width = <8>;
+	no-removable;
+	status = "okay";
+};
+
 &usdhc3 {
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc3>;
-- 
1.7.9.5

